analyzers/tests/SonarAnalyzer.UnitTest/AnalysisContext/SonarAnalysisContextTest.Register.cs

```diff
@@ -163,7 +165,7 @@ public void SonarCompilationStartAnalysisContext_RegisterNodeAction()
         var sut = new SonarCompilationStartAnalysisContext(new(context, DummyMainDescriptor), startContext);
         sut.RegisterNodeAction<SyntaxKind>(CSharpGeneratedCodeRecognizer.Instance, _ => { });
 
-        startContext.AssertExpectedInvocationCounts(expectedNodeCount: 0); // RegisterNodeAction doesn't use DummyCompilationStartAnalysisContext to register but a newly created context
+        startContext.AssertExpectedInvocationCounts(expectedNodeCount: 1);
```

Comment:
I had to change to 1 as we now register to `RoslynAnalysisContext.RegisterSyntaxNodeAction` and not to `SonarAnalysisContent.RegisterNodeAction` anymore in `SonarCompilationStartAnalysisContext `.